bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairControllerHandler.java

```diff
@@ -0,0 +1,1186 @@
+                    }
+                    break;
+                }
+                case CONTROLLER_SETPOINT:
+                case CONTROLLER_HEATMODE: {
+                    PentairHeatStatus status = heatStatusCache.getValue(() -> {
+                        getHeatStatus();
+                    });
+
+                    if (status != null) {
+                        refreshHeatStatusChannels(channelUID, status);
+                    }
+
+                    break;
+                }
+                case CONTROLLER_SCHEDULESTRING:
+                case CONTROLLER_SCHEDULETYPE:
+                case CONTROLLER_SCHEDULECIRCUIT:
+                case CONTROLLER_SCHEDULEDAYS:
+                case CONTROLLER_SCHEDULESTART:
+                case CONTROLLER_SCHEDULEEND: {
+                    int index = getScheduleNumber(groupId) - 1;
+
+                    if (index < 0 || index >= NUM_SCHEDULES) {
+                        return;
+                    }
+
+                    PentairControllerSchedule schedule = schedulesCache[index].getValue(() -> {
+                        getSchedule(index + 1);
+                    });
+
+                    if (schedule != null) {
+                        refreshScheduleChannels(channelUID, schedule);
+                    }
+                    break;
+                }
+                case CONTROLLER_CIRCUITNAME:
+                case CONTROLLER_CIRCUITFUNCTION: {
+                    int circuitNumber = PentairControllerCircuit.getCircuitNumberByGroupID(groupId);
+
+                    if (circuitNumber <= 0 || circuitNumber > NUM_CIRCUITS) {
+                        return;
+                    }
+
+                    PentairControllerCircuit circuit = circuitsCache[circuitNumber - 1].getValue(() -> {
+                        getCircuitNameFunction(circuitNumber);
+                    });
+
+                    if (circuit != null) {
+                        refreshCircuitChannels(channelUID, circuit);
+                    }
+
+                    break;
+                }
+            }
+
+            return;
+        }
+
+        logger.debug("handleCommand: {}", channelUID.getId());
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case CONTROLLER_CIRCUITSWITCH: {
+                if (!(command instanceof OnOffType)) {
+                    break;
+                }
+
+                PentairControllerCircuit circuit = getCircuitByGroupID(groupId);
+                if (circuit == null) {
+                    break;
+                }
+
+                int circuitNum = circuit.id;
+
+                boolean state = ((OnOffType) command) == OnOffType.ON;
+                circuitSwitch(circuitNum, state);
+
+                break;
+            }
+            case CONTROLLER_LIGHTMODE: {
+                if (!(command instanceof StringType)) {
+                    break;
+                }
+
+                String str = ((StringType) command).toString();
```

Comment:
You don't need to cast the command to call `toString`.

Reply:
Can you comment on this?